Add BTS attach and reattach

Pawloland opened the pull request:
Fix #11 
Fix #12

Adrian104 commented:
@Pawloland 
Please edit PR message, because GitHub links "fix" keyword to #11 issue, but not #12:

![image](https://github.com/user-attachments/assets/82602430-9c46-4d42-b6f4-275b0f3661b0)

You need to edit like this:
Fix #11
Fix #12

Besides that, it seems to work well and I didn't find any problems with the code. We have been told during lectures that we should not touch BTS (i.e. we should work only on UE), but you have changed only logging in BTS, so maybe it will be ok.

Pawloland commented:
Modified PR description. 
I changed BTS logging mostly for convenience, to see everything at once in terminal, without zigzagging between a terminal and log files. The default behavior is left as is, so it is more of an extension, which doesn't do any protocol level changes to BTS. I am sure it will be fine. I tried modifying just the logger in COMMON, but unfortunately it wasn't enough as the file is created before logger is even initialized, so changing it in logger caused it to ultimately create syslog files..... but empty ones, because we didn't write to them. 
The current implementation passes a valid file path to a logger, which goes into a void. Logger writes to it normally as for any file, but since it isn't real file we don't pollute file system.
